File: apps/backend/src/EnterpriseAgentOs.Domain/Models/AgentTemplateDto.cs
namespace EnterpriseAgentOs.Domain.Models;

public sealed record AgentTemplateDto(
    Guid Id,
    string Name,
    string Description,
    string Prompt,
    IReadOnlyList<string> Integrations,
    IReadOnlyList<string> Channels,
    bool IsBuiltin);

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/AgentTemplateDto.java
package com.enterpriseagentos.backendjava.domain.features.agents;

import java.util.List;
import java.util.UUID;

public record AgentTemplateDto(UUID id, String name, String description, String prompt, List<String> integrations, List<String> channels, boolean isBuiltin) {
}
